src/main/java/org/jenkins_ci/plugins/flexible_publish/FlexiblePublisher.java

```diff
@@ -112,10 +112,11 @@ public boolean prebuild(final AbstractBuild<?, ?> build, final BuildListener lis
     @Override
     public boolean perform(final AbstractBuild<?, ?> build, final Launcher launcher, final BuildListener listener)
                                                                                                 throws InterruptedException, IOException {
+        boolean wholeResult = true;
         for (ConditionalPublisher publisher : publishers)
             if (!publisher.perform(build, launcher, listener))
-                setResult(build, Result.FAILURE);
-        return true;
+                wholeResult = false;
+        return wholeResult;
```

Comment:
Because it return status of flexible publisher itself. If there is no any bugs in it - it will return true because you must return boolean as method contract. Returning false is bad, because it will fail publisher. Also publisher internals are setting build result directly. I think all answers are here https://github.com/jenkinsci/flexible-publish-plugin/pull/10#issuecomment-74452028